projects/packages/analyzer/scripts/jp-analyze-parallel.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+ini_set( 'memory_limit', '512M' );
+
+require dirname( __DIR__ ) . '/vendor/autoload.php';
+
+use Automattic\Jetpack\Analyzer\Locker;
+use Automattic\Jetpack\Analyzer\Model;
+
+$slurper_path = dirname( __DIR__ ) . '/slurper/plugins';
+
+$jetpack_exclude = array( '.git', 'vendor', 'tests', 'docker', 'bin', 'scss', 'images', 'docs', 'languages', 'node_modules' );
+// $jetpack_new_path = '/Users/brbrr/Developer/a8c/jetpack/projects/plugins/jetpack';
+$jetpack_new_path = '/var/www/html/wp-content/plugins/jetpack';
+$jetpack_old_path = $slurper_path . '/jetpack-production';
+
+class ScanManager {
+	private $count          = 0;
+	private $procs          = array();
+	private $pipez          = array();
+	private $descriptorspec = array(
+		0 => array( 'pipe', 'r' ),
+		1 => array( 'pipe', 'w' ),
+		2 => array( 'file', '/tmp/error-output.txt', 'a' ),
+	);
+
+	private $model;
+
+	public function __construct() {
+		$this->model = new Model();
+	}
+
+	public function start_proc( $folder_name ) {
+			$this->wait_for_procs();
+			echo 'Starting ' . basename( $folder_name ) . "\n";
+			$cmd = 'php ' . dirname( __DIR__ ) . '/scripts/jp-warnings-job.php ' . $folder_name;
+			$this->count++;
+			$process       = proc_open( $cmd, $this->descriptorspec, $pipes );
+			$this->procs[] = $process;
+			$this->pipez[] = $pipes;
+		$this->model->update_process( 0, basename( $folder_name ) );
+
+	}
+
+	public function check_procs() {
+		foreach ( $this->procs as $id => $proc ) {
+			$status = proc_get_status( $proc );
+			if ( $status['pid'] > 0 ) {
+				echo stream_get_contents( $this->pipez[ $id ][1] );
+				$folder_name = explode( ' ', $status['command'] )[2];
+				$this->model->update_process( -1, null, basename( $folder_name ) );
+
+				$this->count--;
+
+				proc_close( $proc );
+				unset( $this->procs[ $id ] );
+				unset( $this->pipez[ $id ] );
+
+				echo 'Done with ' . basename( $folder_name ) . "\n";
+			}
+		}
+	}
+
+	public function wait_for_procs( $count = 5 ) {
+		while ( $this->count > $count ) {
+			$this->check_procs();
+		}
+	}
+
+	public function get_count() {
+		return $this->count;
+	}
+	public function scan( $arr ) {
+		$this->model->update_process( count( $arr ) );
+		foreach ( $arr as $folder_name ) {
+			$this->start_proc( $folder_name );
+		}
```

Comment:
It seems like there should be more logic here, to avoid spawning thousands of processes at once if there are thousands of entries in the array. Spawn up to a limit, then only spawn more once some exit.

Reply:
Ah, yeah, I missed that. :+1:

I do agree that rearranging it might be clearer.